src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1047,17 +1032,31 @@ impl Disambiguator {
         }
     }
 
-    fn display_for(self, path_str: &str) -> String {
+    fn from_res(res: Res) -> Self {
```

Comment:
`from_res`? Why not `impl From<Res> for Disambiguator`?

Reply:
This isn't actually `Into` because it panics on `Res::Err`. So I don't want to make the impl in case someone uses it without looking at the source - it's harder to find trait implementations on pages that rustdoc generates.

Reply:
```suggestion
    /// WARNING: panics on `Res::Err`
    fn from_res(res: Res) -> Self {
```